src/materials/H5MoltresMaterial.C

```diff
@@ -0,0 +1,480 @@
+#include "H5MoltresMaterial.h"
+#include "MooseUtils.h"
+
+// #define PRINT(var) #var
+
+registerMooseObject("MoltresApp", H5MoltresMaterial);
+
+template <>
+InputParameters
+validParams<H5MoltresMaterial>()
+{
+  InputParameters params = validParams<GenericConstantMaterial>();
+  params.addRequiredParam<std::string>("base_file", "The file containing macroscopic XS.");
+  params.addRequiredParam<std::string>("material_key",
+                                       "The file key where the macroscopic XS can be found.");
+  params.addRequiredParam<unsigned>("num_groups",
+                                    "The number of groups the energy spectrum is divided into.");
+  params.addRequiredParam<unsigned>("num_precursor_groups",
+                                    "The number of delayed neutron precursor groups.");
+  params.addCoupledVar(
+      "temperature", 900, "The temperature field for determining group constants.");
+  params.addRequiredParam<MooseEnum>(
+      "interp_type", H5MoltresMaterial::interpTypes(), "The type of interpolation to perform.");
+  return params;
+}
+
+H5MoltresMaterial::H5MoltresMaterial(const InputParameters & parameters)
+  : GenericConstantMaterial(parameters),
+    _temperature(coupledValue("temperature")),
```

Comment:
For instance, all these lines could go in an abstract base class, and wouldn't have to be repeated between MoltresMaterial and H5MoltresMaterial.

Reply:
I apologize, the template specialization would have to be repeated in each file. This is still a minority of the code though.